gas/config/tc-riscv.c

```diff
@@ -2388,6 +2398,11 @@ riscv_ip (char *str, struct riscv_cl_insn *ip, expressionS *imm_expr,
 		  if (!VALID_RVV_VC_IMM (imm_expr->X_add_number))
 		    as_bad (_("bad value for vsetvli immediate field, "
 			      "value must be 0..2047"));
+		  /* Report warning message if the vediv field is set, but the
+		     Zvediv extension isn't enabled.  */
+		  if (!riscv_multi_subset_supports (INSN_CLASS_V_AND_ZVEDIV)
+		      && (imm_expr->X_add_number & (0x3 << 5)))
```

Comment:
We have macros for these magic constants, OP_MASK_VEDIV and OP_SH_VEDIV, which would be better to use.  Though I see that my_getVsetvliExpression isn't using the OP_SH_* macros either and should be fixed, that is probably my fault.

It looks a little odd that my_getVsetvliExpression accepts 'd' without checking INSN_CLASS_V_AND_ZVEDIV.  But you do catch it here.  Maybe add a comment that the caller is expected to check this.  Or maybe move the check into that function.  Shrug.  I don't think it matters.

Reply:
> We have macros for these magic constants, OP_MASK_VEDIV and OP_SH_VEDIV, which 
> would be better to use. Though I see that my_getVsetvliExpression isn't using the OP_SH_* 
> macros either and should be fixed

Yeah it is better to use OP_* macros, I will fix this, and also fix the my_getVsetvliExpression in another commit.

Reply:
> It looks a little odd that my_getVsetvliExpression accepts 'd' without checking 
 INSN_CLASS_V_AND_ZVEDIV. But you do catch it here. Maybe add a comment that the  caller is expected to check this. Or maybe move the check into that function. Shrug. I don't think it matters.

The reason for the beginning is that I want to check the vediv extension after the bad value checking for immediate settings (vsetvli t0, a0, 0x2047).  Hmm, but it seems good to move the checking into the function, and the order of two checking (bad value checking for imm setting and vediv checking) shouldn't matters.